aospy/automate.py

```diff
@@ -0,0 +1,220 @@
+"""Functionality for specifying and cycling through multiple calculations."""
+from __future__ import print_function
+
+import itertools
+import logging
+
+from .calc import Calc, CalcInterface
+
+try:
+    import colorama
+except ImportError:
+    pass
+try:
+    import multiprocess
+except ImportError:
+    pass
+
+
+def _get_obj_default_or_all(name, parent, attr_name, defaults_attr_name):
+    if name == 'all':
+        return getattr(parent, attr_name).values()
+    elif name == 'default':
+        return getattr(parent, defaults_attr_name).values()
+    else:
+        return getattr(parent, attr_name)[name]
+
+
+def _get_mult_objs_by_names(names, parent, attr_name, defaults_attr_name):
+    if isinstance(names, str):
+        names = [names]
+    return [_get_obj_default_or_all(name, parent, attr_name,
+                                    defaults_attr_name) for name in names]
+
+
+def _permute_aux_specs(specs):
```

Comment:
>Indeed, I can confirm this works!

Great!

>For example, right now if you had two Runs with the same name attribute in the list provided, the first one would get overwritten by the second without the user knowing.

Good catch; will address.

>I think you could write this function pretty compactly as a list-comprehension

Yes, will do and the other list comprehensions you suggest